querydsl-examples/querydsl-example-jpa-springboot/src/main/java/com/querydsl/example/jap/springboot/repository/AbstractRepository.java

```diff
@@ -0,0 +1,60 @@
+package com.querydsl.example.jap.springboot.repository;
+
+import com.querydsl.core.types.EntityPath;
+import com.querydsl.core.types.Expression;
+import com.querydsl.example.jap.springboot.model.Identifiable;
+import com.querydsl.jpa.HQLTemplates;
+import com.querydsl.jpa.impl.JPADeleteClause;
+import com.querydsl.jpa.impl.JPAQuery;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.persistence.EntityManager;
+
+public abstract class AbstractRepository<T extends Identifiable> implements Repository<T, Long> {
```

Comment:
Spring-Data users would typically want to see

```java
	public interface MyUserRepository extends JpaRepository<MyUser, Long>, 
  QuerydslPredicateExecutor<MyUser>, QuerydslBinderCustomizer<QMyUser>
```

instead. This doesn't really demonstrate the Spring Data integration, but just how QueryDSL can be used when a `EntityManager` instance is obtainable through any way.